app/main.py

```diff
@@ -1,214 +1,50 @@
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
+from app.knights.param import Knight
+from app.battle.knights_battle import knights_battle
 
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
 
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
+def battle(knights_config: dict) -> dict:
 
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    # BATTLE PREPARATIONS:
+    lancelot = Knight(
+        knights_config["lancelot"]["name"],
+        knights_config["lancelot"]["hp"],
+        knights_config["lancelot"]["power"]
+    )
+    lancelot.calculate_param(knights_config["lancelot"])
+
+    arthur = Knight(
+        knights_config["arthur"]["name"],
+        knights_config["arthur"]["hp"],
+        knights_config["arthur"]["power"]
+    )
+    arthur.calculate_param(knights_config["arthur"])
+
+    mordred = Knight(
+        knights_config["mordred"]["name"],
+        knights_config["mordred"]["hp"],
+        knights_config["mordred"]["power"]
+    )
+    mordred.calculate_param(knights_config["mordred"])
+
+    red_knight = Knight(
+        knights_config["red_knight"]["name"],
+        knights_config["red_knight"]["hp"],
+        knights_config["red_knight"]["power"]
+    )
+    red_knight.calculate_param(knights_config["red_knight"])
```

Comment:
Try simplifying this code, for example, by assigning it all through a loop.


You may ask ‘Why do I need this’, but imagine that you want to change the names of your knights, you will have to change all these keys at the same time.

You can create for loop, where u will create knight instance and then u will store them inside of list/tuple(for example) and then you will call them on fight with list